product_supplier_code_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-17 Eficent Business and IT Consulting Services, S.L.
+#           (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    @api.multi
+    @api.depends('product_id.seller_ids.name',
+                 'product_id.seller_ids.product_code')
+    def _compute_product_supplier_code(self):
+        for line in self:
+            supplier_info = line.product_id.seller_ids.filtered(
```

Comment:
@aheficent Careful with seller_ids as you can have several lines for a product with the same partner (supplierinfo is also supplier pricelist).
This code will crash if several.

You should maybe take supplierinfo[0].